File: UVa00100_The3n1problem.java
package uva;
/* USER: 46724 (sfmunera) */
/* PROBLEM: 36 (100 - The 3n + 1 problem) */
/* SUBMISSION: 07273132 */
/* SUBMISSION TIME: 2009-07-25 23:52:13 */
/* LANGUAGE: 2 */

import java.util.*;

public class UVa00100_The3n1problem
{
    static int cycleLength(int n)
    {
        int cont = 1;
        while (n != 1) {
            if (n%2 == 1)
                n = 3*n+1;
            else
                n = n/2;
            cont++;
        }

        return cont;
    }
    
    public static void main (String args[])  // entry point from OS
    {
        try {
            Scanner in = new Scanner(System.in);
            String input;
            StringTokenizer idata;
            int i, j, max;

            input = in.nextLine();
            while (!input.equals(""))
            {
              idata = new StringTokenizer (input);
              i = Integer.parseInt (idata.nextToken());
              j = Integer.parseInt (idata.nextToken());
              int a;
              int b;

              if (i < j) {
                  a = i;
                  b = j;
              } else {
                  a = j;
                  b = i;
              }
              max = 0;

              for (int k=a; k<=b; k++) {
                  int c = cycleLength(k);
                  if (c > max)
                      max = c;
              }
              System.out.println (i + " " + j + " " + max);
              input = in.nextLine();
            }
        } catch (Exception e) {
            System.exit(0);
        }
    }
}

File: UVa00100_The3n1problem.c
/* USER: 46724 (sfmunera) */
/* PROBLEM: 36 (100 - The 3n + 1 problem) */
/* SUBMISSION: 07273098 */
/* SUBMISSION TIME: 2009-07-25 23:15:40 */
/* LANGUAGE: 1 */

#include <stdio.h>

int cycle(int m)
{
    int i = 1;

    while (m != 1){
	if (m % 2 == 0){
	    m = m/2;
	}
	else{
	    m = 3*m+1;
	}
	i++;
    }
    return i;
}


int main()
{
    int m,n,max,temp;
    int mOriginal,nOriginal;
    int i;

    while (scanf("%d %d\n",&m,&n)==2){
	
	mOriginal = m;
	nOriginal = n;

	if (m > n){
	    temp = m;
	    m = n;
	    n = temp;
	}
	
	max = cycle(m);
	for(i=m+1;i<=n;i++) {
	    temp = cycle(i);
	    if (temp > max)
		max = temp;
	}
	printf("%d %d %d\n",mOriginal,nOriginal,max);
    }
    return 0;
}
